codemodules/match.go

```diff
@@ -0,0 +1,34 @@
+package codemodules
+
+import (
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/dtlabels"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func MatchForNamespace(codeModuleDynakubes []dynatracev1alpha1.DynaKube, namespace *corev1.Namespace) (*dynatracev1alpha1.DynaKube, error) {
+	var matchingModules []dynatracev1alpha1.DynaKube
+
+	for _, codeModule := range codeModuleDynakubes {
+		matching, err := dtlabels.IsMatching(
```

Comment:
Just to confirm since I just ran into it, `metav1.LabelSelectorAsSelector()` does some of the logic here, e.g.,

```go
sel, err := metav1.LabelSelectorAsSelector(&dk.Spec.CodeModules.Selector)
if sel.Matches(labels.Set(dk.Labels)) {
  // Do something.
}
```

Reply:
changed and removed redundant code.
Thank you for that btw, I looked for something like that, but couldn't find anything.